### PR TITLE
Enable partial reversals

### LINKED ISSUE #198: Allow partial refunds
clicking “Reverse Transaction” will now go to a reversal screen that defaults to 100% reversal but allows you to modify the values if you need to do a partial.  Once sent, the rest of the process will be the same as before.

Future state -- instead of a single reversal, a transaction can have many reversals.  Should be easy from the txn page to link to the reversal and backwards.

### PR DESCRIPTION
**Pull request by sprucely**
One thing I wasn't sure about was if a small fee rounds to less than 1 cent after the percentage calculation. Not sure if it will ever hit that edge case, but I opted to skip those transfers.

Closes #198

### CODE REVIEW

**Review by jSylvestre — CHANGES_REQUESTED** [review]
You need to sum up all the debits and credits, and make sure they are equal

**Comment by sprucely** [comment]
> You need to sum up all the debits and credits, and make sure they are equal

I'll add it. But why? I assume that logic wasn't there in the first place because it was reversing transfers that already existed. All this does is reverse a percentage of those already existing transfers.

**Comment by jSylvestre** [comment]
> > You need to sum up all the debits and credits, and make sure they are equal
> 
> I'll add it. But why? I assume that logic wasn't there in the first place because it was reversing transfers that already existed. All this does is reverse a percentage of those already existing transfers.

Before everything was just switched debit to credit. They already had to match.
If you are using % now,  there is the possibility with that and rounding that the amounts will no longer match.

**Review by srkirkland — COMMENTED** [review]
@sprucely  Without testing a complex case, I'm a little surprised that the percentage rounding will work in all cases.  I would think you'd end up with some off by `$0.01` errors pretty easily.  If you've tested though, I think the other thing I'd say is to make the UX a little better, maybe at least add the [$] prefix to the input tag.  Maybe add a label into a card.  And a pinch of right margin so the textbox doesn't run into the button

File: Sloth.Core/SlothDbContext.cs [hunk 1]
```diff
@@ -84,7 +84,7 @@ public async Task<string> GetNextKfsTrackingNumber(DbTransaction transaction = n
             }
         }
 
-        public async Task<string> GetNextDocumentNumber(DbTransaction transaction = null)
+        public virtual async Task<string> GetNextDocumentNumber(DbTransaction transaction = null)
```
Comment (jSylvestre) [inline Sloth.Core/SlothDbContext.cs:87]: virtual was just so the db can be mocked out, correct?

File: Sloth.Test/Helpers/CreateValidEntities.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            rtValue.Description = $"Transaction {counter}";
+            rtValue.TransactionDate = DateTime.Now;
+            rtValue.Transfers = transfers;
+            rtValue.SetStatus(status ?? TransactionStatuses.Completed);
+            return rtValue;
+        }
+
+        public static Transfer Transfer(Transfer.CreditDebit direction, decimal amount, int? counter)
+        {
+            var rtValue = new Transfer();
```
Comment (jSylvestre) [inline Sloth.Test/Helpers/CreateValidEntities.cs:38]: I'd also add a description, then check the correct ones were updated. But not a big deal.

File: Sloth.Core/SlothDbContext.cs (hunk 1 above)
Comment (sprucely) [reply Sloth.Core/SlothDbContext.cs:87]: Yes.

**Review by jSylvestre — APPROVED** [review]

**Pull request merged**